### PR TITLE
[Correlation_916] Bipul/Aditi | Add CorrelationId to trace request flow

### PR DESCRIPTION
**Pull request by imaditiagg**

### CODE REVIEW

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- src/In.ProjectEKA.HipService/Gateway/GatewayClient.cs  1
- src/In.ProjectEKA.HipService/DataFlow/DataFlowController.cs  6
- src/In.ProjectEKA.HipService/Common/HttpRequestHelper.cs  1
- src/In.ProjectEKA.HipLibrary/Patient/Model/TraceableDataRequest.cs  1
         

Clones added
============
- src/In.ProjectEKA.HipLibrary/Patient/Model/TraceableDataRequest.cs  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/hip-service/pullRequest?prid=6116067&bid=20016189)

File: src/In.ProjectEKA.HipService/Common/HttpRequestHelper.cs [hunk 1]
```diff
@@ -33,13 +35,15 @@ public static class HttpRequestHelper
                 httpRequestMessage.Headers.Add(HeaderNames.Authorization, token);
             if (cmSuffix != null)
                 httpRequestMessage.Headers.Add("X-CM-ID", cmSuffix);
+            if (correlationId != null)
```
Comment (ganesanarun) [inline src/In.ProjectEKA.HipService/Common/HttpRequestHelper.cs:38]: Codacy found an issue: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/gh/ProjectEKA/hip-service/pullRequest?prid=6116067)

File: src/In.ProjectEKA.HipService/Gateway/GatewayClient.cs [hunk 2]
```diff
@@ -45,7 +45,8 @@ public virtual async Task<Option<string>> Authenticate()
                     Method = HttpMethod.Post,
                     Content = new StringContent(json, Encoding.UTF8, MediaTypeNames.Application.Json)
                 };
-
+                if (correlationId != null)
```
Comment (ganesanarun) [inline src/In.ProjectEKA.HipService/Gateway/GatewayClient.cs:48]: Codacy found an issue: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/gh/ProjectEKA/hip-service/pullRequest?prid=6116067)

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- src/In.ProjectEKA.HipService/Common/HttpRequestHelper.cs  1
- src/In.ProjectEKA.HipService/DataFlow/DataFlowController.cs  6
- src/In.ProjectEKA.HipLibrary/Patient/Model/TraceableDataRequest.cs  1
- test/In.ProjectEKA.HipServiceTest/DataFlow/Builder/TestBuilder.cs  1
- src/In.ProjectEKA.HipService/Gateway/GatewayClient.cs  1
         

Clones added
============
- test/In.ProjectEKA.HipServiceTest/Gateway/GatewayClientTest.cs  4
- src/In.ProjectEKA.HipLibrary/Patient/Model/TraceableDataRequest.cs  3
- test/In.ProjectEKA.HipServiceTest/DataFlow/Builder/TestBuilder.cs  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/hip-service/pullRequest?prid=6116067&bid=20016189)

File: src/In.ProjectEKA.HipService/Common/HttpRequestHelper.cs (hunk 1 above)
Comment (ganesanarun) [inline src/In.ProjectEKA.HipService/Common/HttpRequestHelper.cs:38]: Codacy found an issue: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/gh/ProjectEKA/hip-service/pullRequest?prid=6116067)

**Review by mddubey — APPROVED** [review]

File: src/In.ProjectEKA.HipService/Common/HttpRequestHelper.cs (hunk 1 above)
Comment (ganesanarun) [inline src/In.ProjectEKA.HipService/Common/HttpRequestHelper.cs:38]: Codacy found an issue: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/gh/ProjectEKA/hip-service/pullRequest?prid=6116067)

File: src/In.ProjectEKA.HipService/Gateway/GatewayClient.cs (hunk 2 above)
Comment (ganesanarun) [inline src/In.ProjectEKA.HipService/Gateway/GatewayClient.cs:48]: Codacy found an issue: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/gh/ProjectEKA/hip-service/pullRequest?prid=6116067)

**Pull request merged**